File: src/main/java/sample/SampleViewModel.java
package sample;

import de.saxsys.mvvmfx.ViewModel;
import de.saxsys.mvvmfx.utils.commands.Action;
import de.saxsys.mvvmfx.utils.commands.Command;
import de.saxsys.mvvmfx.utils.commands.DelegateCommand;
import javafx.beans.property.ReadOnlyStringProperty;
import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.StringProperty;

public class SampleViewModel implements ViewModel {
    private final StringProperty halloLabelText = new SimpleStringProperty("hallo1");
    private final Command buhCommand;

    public SampleViewModel() {
        buhCommand = new DelegateCommand(() -> new Action() {
            @Override
            protected void action() {
                halloLabelText.set("hallo buh");
            }
        });
    }

    public ReadOnlyStringProperty halloLabelTextProperty() {
        return halloLabelText;
    }

    public Command getBuhCommand() {
        return buhCommand;
    }
}

File: src/main/kotlin/sample/SampleViewModel.kt
package sample

import de.saxsys.mvvmfx.ViewModel
import de.saxsys.mvvmfx.utils.commands.Action
import de.saxsys.mvvmfx.utils.commands.Command
import de.saxsys.mvvmfx.utils.commands.DelegateCommand
import javafx.beans.property.SimpleStringProperty
import javafx.beans.property.StringProperty

public class SampleViewModel : ViewModel {
    private val halloLabelInitialText = "hallo1"

    public val halloLabelText: StringProperty = SimpleStringProperty(halloLabelInitialText)
    public val buhCommand: Command = DelegateCommand({ object : Action() {
        override fun action() {
            halloLabelText.set("hallo buh")
        }
    } }, halloLabelText.isEqualTo(halloLabelInitialText))
}
